### PR TITLE
[SPARK-29888][SQL] new interval string parser shall handle numeric with only fractional part

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

Current string to interval cast logic does not support i.e. cast('.111 second' as interval) which will fail in SIGN state and return null, actually, it is 00:00:00.111. 
```scala
-- !query 63
select interval '.111 seconds'
-- !query 63 schema
struct<0.111 seconds:interval>
-- !query 63 output
0.111 seconds


-- !query 64
select cast('.111 seconds' as interval)
-- !query 64 schema
struct<CAST(.111 seconds AS INTERVAL):interval>
-- !query 64 output
NULL
````
### Why are the changes needed?

bug fix.
### Does this PR introduce any user-facing change?

no
### How was this patch tested?

add ut

### CODE REVIEW

**Comment by yaooqinn** [comment]
cc @cloud-fan @MaxGekk, thanks in advance. this pr is raised via comment @ https://github.com/apache/spark/pull/26491#discussion_r346151299

**Review by cloud-fan — APPROVED** [review]

**Comment by yaooqinn** [comment]
retest this please

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala [hunk 1]
```diff
@@ -496,6 +496,14 @@ object IntervalUtils {
           state = TRIM_BEFORE_SIGN
         case TRIM_BEFORE_SIGN => trimToNextState(b, SIGN)
         case SIGN =>
+          currentValue = 0
+          fraction = 0
+          // We preset next state from SIGN to TRIM_BEFORE_VALUE. If we meet '.' in the SIGN state,
```
Comment (MaxGekk) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala]: The comment is useless from my point of view. It describes what is obvious from the code but does not say why.

Comment (cloud-fan) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala]: agree. @yaooqinn can you refine the comments to explain why? We should mention that it's to support `.1 second`

Comment (yaooqinn) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala]: OK

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Comment by yaooqinn** [comment]
thanks for merging

**Pull request closed without merging**